src/code/z_message_nes.c

```diff
@@ -288,8 +290,1585 @@ void Message_LoadOwlWarpTextNES(PlayState* play, s32* offset, f32* arg2, s16* de
     *arg2 = f;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_message_nes/func_80159438.s")
+void Message_GetTimerDigitsNES(OSTime time, s16* digits) {
+    OSTime t = time;
+
+    // 6 minutes
+    digits[0] = t / SECONDS_TO_TIMER(360);
+    t -= (digits[0] * SECONDS_TO_TIMER(360));
+
+    // minutes
+    digits[1] = t / SECONDS_TO_TIMER(60);
+    t -= (digits[1] * SECONDS_TO_TIMER(60));
+
+    digits[2] = '\'';
+
+    // 10 seconds
+    digits[3] = t / SECONDS_TO_TIMER(10);
+    t -= digits[3] * SECONDS_TO_TIMER(10);
+
+    // seconds
+    digits[4] = t / SECONDS_TO_TIMER(1);
+    t -= digits[4] * SECONDS_TO_TIMER(1);
+
+    digits[5] = '"';
+
+    // 100 milliseconds
+    digits[6] = t / SECONDS_TO_TIMER_PRECISE(0, 10);
+    t -= digits[6] * SECONDS_TO_TIMER_PRECISE(0, 10);
+
+    // 10 milliseconds
+    digits[7] = t;
+
+    digits[0] += '0';
+    digits[1] += '0';
+    digits[3] += '0';
+    digits[4] += '0';
+    digits[6] += '0';
+    digits[7] += '0';
+}
+
+// Probably color structs instead of Vec3s... whoops
+s16 D_801D07DC[][3] = {
+    { 255, 120, 0 },  { 70, 255, 80 },   { 80, 110, 255 },  { 255, 255, 30 },
+    { 90, 180, 255 }, { 210, 100, 255 }, { 170, 170, 170 }, { 255, 130, 30 },
+};
+
+s16 D_801D080C[][3] = {
+    { 255, 60, 60 },  { 70, 255, 80 },   { 80, 90, 255 },   { 255, 255, 50 },
+    { 80, 150, 255 }, { 255, 150, 180 }, { 170, 170, 170 }, { 255, 130, 30 },
+};
+
+s16 D_801D083C[][3] = {
+    { 255, 60, 60 },  { 70, 255, 80 },   { 80, 90, 255 },   { 255, 255, 50 },
+    { 80, 150, 255 }, { 255, 150, 180 }, { 180, 180, 200 }, { 255, 130, 30 },
+};
+
+s16 D_801D086C[][3] = {
+    { 195, 0, 0 },    { 70, 255, 80 },   { 80, 90, 255 },   { 255, 255, 50 },
+    { 80, 150, 255 }, { 255, 150, 180 }, { 170, 170, 170 }, { 255, 130, 30 },
+};
+
+s16 D_801D089C[][3] = {
+    { 255, 60, 60 },  { 110, 170, 255 }, { 80, 90, 255 },   { 255, 255, 50 },
+    { 80, 150, 255 }, { 255, 150, 180 }, { 170, 170, 170 }, { 255, 130, 30 },
+};
+
+u8 D_801D08CC[] = {
+    0x02, 0x01, 0x03, 0x06, 0x06, 0x06, 0x03, 0x03, 0x03, 0x03, 0x01, 0x06,
+};
+
+#ifdef NON_MATCHING
```

Comment:
Is there a decomp.me of this? I usually put that in a comment in the NON_MATCHING section.